boards/openhwgroup/cv32a6_arty_a7_100/Kconfig.cv32a6_arty_a7_100

```diff
@@ -0,0 +1,5 @@
+# Copyright 2024 CISPA Helmholtz Center for Information Security gGmbH
+# SPDX-License-Identifier: Apache-2.0
+config BOARD_CV32A6_ARTY_A7_100
```

Comment:
split this into commits, not going to review 77 files spanning multiple boards, multiple socs and even samples in a single commits

Reply:
I have removed the boards and samples from this PR and created new PRs:
- https://github.com/zephyrproject-rtos/zephyr/pull/81019
- https://github.com/zephyrproject-rtos/zephyr/pull/81015
- https://github.com/zephyrproject-rtos/zephyr/pull/81018
- https://github.com/zephyrproject-rtos/zephyr/pull/81017
I have kept the 32-bit and the 64-bit variants of the same board in the same PR, as they are otherwise identical.

Reply:
Have this PR have a soc (in one commit), drivers (each in different commits) and one board (in another commit), then once that is merged you can add additional boards in a new PR each with their own commit

Reply:
The Ethernet subsystem is not specific to this hardware (e.g., it is also used on ARM boards), so I will leave it as separate PRs.
This PR now has one commit with the SoC and one with a board in 32 and 64 bit variants.